components/google_sheets/actions/upsert-row/upsert-row.mjs

```diff
@@ -0,0 +1,154 @@
+import { v4 as uuid } from "uuid";
+import googleSheets from "../../google_sheets.app.mjs";
+
+/**
+ * This action performs an upsert operation, similar to the MySQL `INSERT INTO ... ON DUPLICATE KEY
+ * UPDATE` operation, on a Google Spreadsheet. If a row in the spreadsheet has `value` in `column`
+ * (i.e., a duplicated key), that row is updated. Otherwise, a new row is appended to the
+ * spreadsheet.
+ *
+ * To determine if and where a duplicate key exists in the spreadsheet, this action uses [Google
+ * Sheet's `MATCH` function](https://support.google.com/docs/answer/3093378) in a new temporary
+ * hidden worksheet. Uses roughly the method described in [this stackoverflow
+ * answer](https://stackoverflow.com/a/49439220) and [this GitHub
+ * comment](https://github.com/PipedreamHQ/pipedream/issues/1824#issuecomment-949940177).
+ */
+export default {
+  key: "google_sheets-upsert-row",
+  name: "Upsert Row",
+  description: "Upsert a row of data in a Google Sheet",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    googleSheets,
+    drive: {
+      propDefinition: [
+        googleSheets,
+        "watchedDrive",
+      ],
+      description: "",
+      optional: true,
+    },
+    sheetId: {
+      propDefinition: [
+        googleSheets,
+        "sheetID",
+        (c) => ({
+          driveId: googleSheets.methods.getDriveId(c.drive),
+        }),
+      ],
+    },
+    sheetName: {
+      propDefinition: [
+        googleSheets,
+        "sheetName",
+        (c) => ({
+          sheetId: c.sheetId,
+        }),
+      ],
+    },
+    insert: {
+      propDefinition: [
+        googleSheets,
+        "cells",
+      ],
+      label: "Insert",
+      description: "Insert statement: the row data you want to add to the Google sheet if the key *doesn't* exist. If the key *does* exist and **Update** is not set, the row will be updated using this array. Use structured mode to enter individual cell values. Disable structured mode to pass an array with each element representing a cell/column value (e.g. `{{ [5, \"test\"] }}`).",
```

Comment:
Just a question. Shouldn't we have like a way to set these strings in a more developer friendly reading format such as having like an array and then joining the paragraphs. Because sometimes it's a bit painful to go all the way up to the end of the description scrolling horizontally to get the whole text. I mean something like
```suggestion
      description: [
          "Insert statement: the row data you want to add to the Google sheet if the key *doesn't* exist.",
          "If the key *does* exist and **Update** is not set, the row will be updated using this array.",
          "Use structured mode to enter individual cell values.",
          "Disable structured mode to pass an array with each element representing a",
          "cell/column value (e.g. `{{ [5, \"test\"] }}`).",
        ].join(" "),
```

Reply:
I added a [`toSingleLineString` function](https://github.com/PipedreamHQ/pipedream/blob/11c2f3c87747e1d3d9b29ce63c45d63e4795612c/components/google_sheets/utils.mjs#L25) in `google_sheets.utils.mjs`, and used it to split the `description` for `insert` and `updates` into multiple lines. The code for the `toSingleLineString` function is from [`aws/sources/common/utils.mjs`](https://github.com/PipedreamHQ/pipedream/blob/11c2f3c87747e1d3d9b29ce63c45d63e4795612c/components/aws/sources/common/utils.mjs#L34).